### ISSUE TITLE
Beast2: environment is missing libXrender.so

### ISSUE THREAD

**Issue by lgorenstein** [bioconda/bioconda-recipes#27854]
Dear Bioconda team, when running Beast2 from a Biocontainers' container, the `beauti` GUI fails to start due to a missing `libXrender.so` library:
```
$ singularity run beast2:2.6.3--hf1b8bbb_0.sif beauti
WARNING: Skipping mount /var/singularity/mnt/session/etc/resolv.conf [files]: /etc/resolv.conf doesn't exist in container
Exception in thread "main" java.lang.UnsatisfiedLinkError: /usr/local/jre/lib/amd64/libawt_xawt.so: libXrender.so.1: cannot open shared object file: No such file or directory
```

Of course this is never a problem when running from a conda environment natively (the environment can always reach out to the host's library - in my case on CentOS7 it's in `/usr/lib64`), but it does become a problem when running a containerized version.  I could kind of work around it by binding the host's `/usr/lib64` into the container, but this feels like too dirty of a trick to be a real solution.

Would it be possible to add `libXrender` into the environment?  This should solve the issue on the spot.

**Comment by dlaehnemann** [comment]
I think this is probably down to depending on an old version of `openjdk`, namely this one:
https://github.com/conda-forge/openjdk-feedstock/blame/7fa1a1d1735b83ced0a6873463c4ad74e0407c50/recipe/meta.yaml

So you might have to dig into the archives there, to see why libXrender is not included. Or try updating the recipe to a newer version of `openjdk` (there's at least one newer openjdk version under the major version `8.` on conda-forge).

**Comment by lgorenstein** [comment]
Thanks, this could be it.

And herein lies the crux - note that I was not building a container afresh, I was `singularity pull`-ing an existing image from the Biocontainers registry.  So the crux was to include `libXrender.so` (or said newer `openjdk`) into the Biocontainer... to which Biocontainers' team general response is "It's a Bioconda thing, so we just use the upstream conda recipe as is" :)

I am no longer in need of this container, but out of curiosity I just pulled the latest version, and it still suffers from the same issue:
```
$ singularity pull docker://quay.io/biocontainers/beast2:2.6.3--he65b2d3_2

$ singularity exec ./beast2_2.6.3--he65b2d3_2.sif beauti
Exception in thread "main" java.lang.UnsatisfiedLinkError: /usr/local/jre/lib/amd64/libawt_xawt.so: libXrender.so.1: cannot open shared object file: No such file or directory
[...]
```

Again, the main problem here is containerization (what's included in the container). When the conda environment is installed on a "normal" system, chances are all X-rendering libraries are already present implicitly, so to speak, so conda sees them and is happy. But for containerized builds they need to be included into the container explicitly, which does not seem to happen.

**Comment by dlaehnemann** [comment]
Yeah, the basic problem is an implicit vs. an explicit dependency. `libXrender.so` is also available via conda-forge, but this old version of `openjdk` simply doesn't declare it as a dependency. The best solution would probably be to patch the old `openjdk` version on conda-forge to actually state that dependency explicitly in its `meta.yaml`. But I don't have the bandwidth to do that, without needing it myself...

Also, newer versions of beast2 depend on newer versions of Java, but haven't been packaged for bioconda, yet. So that could be another option. Although I'm not sure at which point `openjdk` was "fixed" in this regard. It might be that `openjdk` `v9.x` still has this undeclared dependency, I didn't check.

**Comment by lgorenstein** [comment]
Back when I started this thread, I was thinking that the easiest practical solution (although arguably not the cleanest) would have been adding `libXrender.so` dependency to the beast2 Bioconda recipe.  This way it would get added into the container no matter what.

Nowadays, we should probably wait and see how new version of Beast2 is packaged, and revisit this thought if it still doesn't.

**Comment by dlaehnemann** [comment]
Yes, if all else fails, this would be the quick and dirty solution. And with the complex dependencies and Java version restrictions, this would probably be easier than patching the old openjdk version. But unless someone actually needs this, I wouldn't put in time for either effort. Just wanted to document my thoughts -- maybe someone who does need at some point, at least can build on our discussion here. :crossed_fingers:
